sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationTokenCredential.java

```diff
@@ -41,6 +41,8 @@
     public CommunicationTokenCredential(String token) {
         Objects.requireNonNull(token, "'token' cannot be null.");
         setToken(token);
+        CommunicationTokenRefreshOptions tokenRefreshOptions = new CommunicationTokenRefreshOptions(null, false, token);
+        refreshTimeBeforeTokenExpiry = tokenRefreshOptions.getRefreshTimeBeforeTokenExpiry();
```

Comment:
since `refreshTimeBeforeTokenExpiry` is being used in the `setToken` method, I'd do the actions in the constructor in a different order - similar to other constructors, meaning:
1) set the `refreshTimeBeforeTokenExpiry` first
2) call `setToken` second